maths/special_numbers/happy_number.py

```diff
@@ -0,0 +1,39 @@
+def is_happy_number(number: int) -> bool:
+    """
+    Check if a number is a happy number.
+    https://en.wikipedia.org/wiki/Happy_number
+     A happy number is defined by the following process:
+    1. Starting with any positive integer, replace the number by
+       the sum of the squares of its digits.
+    2. Repeat the process until the number equals 1 (happy) or
+       it loops endlessly in a cycle (not happy).
+
+    Args:
+        n (int): The number to check for happiness.
+
+    Returns:
+        bool: True if the number is a happy number, False otherwise.
+
+    Examples:
+    >>> is_happy_number(19)
+    True
+    >>> is_happy_number(4)
+    False
+    >>> is_happy_number(23)
+    True

```

Comment:
Please add tests the break things.
```suggestion
    True
    >>> is_happy_number(0)
    ?
    >>> is_happy_number(-19)
    ?
    >>> is_happy_number(19.1)
    ?
    >>> is_happy_number("Happy")
    ?
```

Reply:
It is not resolved until the tests pass.

Reply:
Please put the requested tests back in place and have your doctests catch the raised ValueErrors just like numerous other algorithms in this repository already do.  If you are unwilling to follow directions then please close this pull request because there are 160+ other PRs that can be reviewed instead.